### PR TITLE
replace is_incomplete with has_complete_metadata for less confusing double negatives

### PR DESCRIPTION
**Pull request by eviljeff**
Also realised that versions/forms.py is unused so deleted it.

### CODE REVIEW

**Comment by eviljeff** [comment]
another part of #3469

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -1271,18 +1271,24 @@ def is_unreviewed(self):
     def is_public(self):
         return self.status == amo.STATUS_PUBLIC and not self.disabled_by_user
 
-    def is_incomplete(self):
-        # Don't hassle add-ons with only unlisted versions.
-        if not self.is_listed:
-            return False
-
+    def has_complete_metadata(self, for_channel=None):
```
Comment (diox) [inline src/olympia/addons/models.py]: You're never using the `for_channel` argument in this method or the one below. Remove it ?

File: src/olympia/addons/models.py [hunk 2]
```diff
@@ -1271,18 +1271,24 @@ def is_unreviewed(self):
...
             channel=amo.RELEASE_CHANNEL_LISTED)
-        return not (
-            self.all_categories and
-            self.summary and
-            (not latest_version or latest_version.license)
-        )
+        return [
+            self.all_categories,
+            self.summary,
+            (not latest_version or latest_version.license),
```
Comment (diox) [inline src/olympia/addons/models.py]: It was already there, but what about just `latest_version and latest_version.license` ? I find it less confusing.

Comment (diox) [inline src/olympia/addons/models.py]: Ah, I get it now. It's confusing...

File: src/olympia/addons/models.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/addons/models.py]: https://github.com/mozilla/addons-server/pull/3926/commits/283d1a9ceb7ade3d9fc8af9b22f2c6cba6d760f4

**Pull request merged**